enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,80 @@
+
+---
+title: SHA-Bundle-Digest
+authors:
+- "@theishshah"
+reviewers:
+- @camilamacedo86 
+- @joelanford 
+- @jmrodri
+- @ryantking 
+- @tlwu2013 
+approvers:
+- @tlwu2013 
+- @jmrodri
+- @joelanford 
+creation-date: 2022-01-14
+last-updated: 2022-01-14
+status:implementable
+---
+
+# Title
+Add support for SHA-based Digest for Images
+
+## Summary
+
+Currently, operators generated by `operator-sdk` will store and retrieve container images based on the image name and the image tag in the bundle. This EP adds support to `operator-sdk` such that images can be stored and retrieved in the bundle with an image name and an image digest as an optional replacement for a tag. Image digests are SHA-256 sums that are hashed from the image layer therefore uniquely identifying one specific image.
+
+## Motivation
+
+Users in air-gapped (disconnected) environments are currently unable to get images based on tags, and can only access local registries based on the digest.
+
+This set of changes will allow SDK users in air-gapped (disconnected) environments to generate bundles with the image digest. This change is necessary due to non-deterministic behavior with images identified via tags, this issue specifically occurs in the OCP registry due to how the digest is generated with tag name as inputs.
+
+### Goals
+
+-   Give users the ability to the enable digest-based image resolution via a CLI flag when generating bundles.
+-   Allow users to specify additional images to resolve digests via an environment variable.
+-   Add the ability to resolve image digests from image tag dynamically using local docker registry.
+-   OPEN QUESTION: Store digest and tag mappings in file so that user-written code can resolve tags to digest.
+
+### Non-Goals
+
+-   Migrate existing `operator-sdk` based operators from image tags to image digests.
+-   Identify locations in user code that reference images by tag.
+
+## Proposal
+
+-   Add `--image-digests` flag to `make bundle` in order to enable this feature to be run. 
+
+-   During bundle generation, discover all environment variables that begin with `RELATED_IMAGE_` then resolve the tags to digests. These variables should be set in `config/manager/manager.yaml` as suggested in [https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions](https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions). 
```

Comment:
I think we are missing a bullet on the CLI flag for the `operator-sdk generate bundle` command. I think the general flow of the proposal should be:

```
- Add `--use-image-digests` to `operator-sdk generate bundle` command. When enabled, the CLI will:
  - Search the environment defined in `manager.yaml` for all variables that begin with `RELATED_IMAGE_` to build a list of all images that need digests in the output cluster CSV.
  - Resolve the digests for the operator image and any additional related images it finds in the environment. 
  - Output the images with digests instead of tags in the cluster CSV.
- Add a variable to the `Makefile` called `USE_IMAGE_DIGESTS`.
```

Now the section is focused on the specific changes to how the user interacts with the software and how its functionality changes based on that user interaction.